### PR TITLE
SwayNotificationCenter: init at 0.3

### PR DESCRIPTION
**Pull request by berbiche**
###### Motivation for this change

Package https://github.com/ErikReider/SwayNotificationCenter.

I have been using this notification daemon for few weeks now.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/swaynotificationcenter/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "v${version}";
+    hash = "sha256-gXo/V2FHkHZBRmaimqJCzi0BqS4tP9IniIlubBmK5u0=";
+  };
+
+  nativeBuildInputs = [ gobject-introspection meson ninja pkg-config vala wrapGAppsHook ];
+
+  buildInputs = [ dbus dbus-glib gdk-pixbuf glib gtk-layer-shell gtk3 json-glib libhandy librsvg ];
+
+  meta = with lib; {
+    description = "A simple notification daemon with a gui built for Sway";
```
Comment (coolshaurya) [inline pkgs/applications/misc/swaynotificationcenter/default.nix:24]: nitpick 😄: gui should be uppercase "GUI"

**Comment by mvnetbiz** [comment]
I think it's also preferred to have one package per line.

File: pkgs/applications/misc/swaynotificationcenter/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/swaynotificationcenter/default.nix:24]:
```suggestion
    description = "Simple notification daemon with a GUI built for Sway";
```

**Comment by berbiche** [comment]
> I think it's also preferred to have one package per line.

Fixed

> gui should be uppercase "GUI"

Fixed

Thanks for the reviews!

**Comment by mvnetbiz** [comment]
LGTM on the package. To get it to work I have to add it to 3 lines to get dbus to register and without it installed to systemPackages I get the error `COULD NOT FIND CSS FILE! REINSTALL THE PACKAGE!` Can you add a module in `nixos/modules/programs/`?
```nix
{
    environment.systemPackages = [ swaynotificationcenter ];
    services.dbus.packages = [ swaynotificationcenter ];
    systemd.packages = [ swaynotificationcenter ];
}
```

**Comment by berbiche** [comment]
> I get the error COULD NOT FIND CSS FILE! REINSTALL THE PACKAGE!

``` console
$ strace -e openat,open,stat swaync |& swaync
<snip>
openat(AT_FDCWD, "/etc/profiles/per-user/nicolas/etc/xdg/swaync/style.css", O_RDONLY) = 9
openat(AT_FDCWD, "/etc/profiles/per-user/nicolas/etc/xdg/swaync/config.json", O_RDONLY) = 9
```
This is the log I have with the package in my `home.packages` (under Home Manager).
It does the same thing as `environment.systemPackages` in terms of linking files and folders.

I could wrap the swaync binary and append the package's `$out/etc/xdg` to `$XDG_DATA_DIRS`.
It would probably resolve your error.

> Can you add a module in nixos/modules/programs/?

I already have a working module for Home Manager (that I have not yet contributed).
I don't intend to create a module for NixOS because I personally don't manage tools like this with NixOS, so feel free to submit one.

**Comment by mvnetbiz** [comment]
I don't have /etc/profiles/per-user. Is that from home manager nixos module? I am using home-manager too, so it would be okay if it works that way, but I don't use the home manager nixos module.

**Pull request merged**